scripts/entity_connexion.py

```diff
@@ -0,0 +1,189 @@
+
+import numpy as np
+import pandas as pd
+import networkx as nx
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+
+from eutl_orm import DataAccessLayer
+from attic.connection_settings import connectionSettings
+from eutl_orm import Installation, Account, AccountHolder
+
+dal = DataAccessLayer(**connectionSettings)
+session = dal.session
+
+
+class EntityConnexion:
+    def __init__(self, entity_type, entity_id, period=(None, None)):
+        if entity_type in ['Account', 'Company', 'AccountHolder']:
+            self.entity_type = entity_type
+        else:
+            print('entity_type needs to be one of Account, Company or AccountHolder')
+            raise ValueError
+        self.entity_id = entity_id
+        self.period = period
+        self.accounts = None
+        self.transactions = None
+        self.entity_name = None
+
+        self.get_accounts()
+        self.get_transactions()
+        self.plot_arrows()
+
+    def get_accounts(self):
+        # look up list of accounts and return it
+        if self.entity_type == 'Account':
+            self.accounts = session.query(Account).filter(Account.id == self.entity_id).all()
+            self.entity_name = self.accounts[0].name
+        elif self.entity_type == 'AccountHolder':
+            self.accounts = session.query(Account).filter(Account.accountHolder_id == self.entity_id).all()
+            self.entity_name = session.query(AccountHolder).filter(AccountHolder.id == self.entity_id).first().name
+        elif self.entity_type == 'Company':
+            self.accounts = session.query(Account).filter(Account.companyRegistrationNumber == self.entity_id).all()
+            self.entity_name = ''
+
+        print('You are working with {} with id {} ({}) - {} related accounts'.format(self.entity_type,
```

Comment:
(Cosmétique) - tu pourrais aussi utiliser des f-strings, c'est plus facile à lire.

A part ca, la phrase est pas super sybilline quand tu bosses avec un "Account":

`You are working with Account with id 493 (AT FHKW Spittelau Fernwärme Wien) - 1 related accounts`

Tu pourrais peut-être afficher une phrase différente selon qu'on à affaire à un Account ou un AccountHolder/Firm ?

Reply:
Done.